content/spaces/cursors.textile

```diff
@@ -0,0 +1,164 @@
+---
+title: Live cursors
+meta_description: "Track the positions of cursors within a space."
+languages:
+  - javascript
+---
+
+The live cursors feature enables you to track the cursors of members within a space in realtime.
+
+Cursor events are emitted whenever a member moves their mouse within a space. Live cursor updates are not available as part of the "space state":/spaces/space#subscribe and must be subscribed to using "@cursors.subscribe()@":#subscribe.
+
+A maximum of 50 members using live cursors per space is recommended for optimal performance.
+
+<aside data-type='further-reading'>
+<p>Live cursors build upon existing functionality available in Ably's Core SDKs. Read about "space management":/spaces/space#management to understand the features and underlying channels associated with using live cursors.</p>
+</aside>
+
+h2(#set). Set cursor position
+
+Set the position of a member's cursor using the @set()@ method. A position must contain an X-axis value and a Y-axis value to set the cursor position on a 2D plane. It may also contain additional information about the cursor, such as the color other members should display it as. Calling @set()@ will emit a cursor event so that other members are informed of the cursor movement in realtime.
+
+The @set()@ method takes the following parameters:
+
+|_. Parameter |_. Description |_. Type |
+| position.x | The position of the member's cursor on the X-axis. | Number |
+| position.y | The position of the member's cursor on the Y-axis. | Number |
+| data | An optional arbitrary JSON-serializable object containing additional information about the cursor, such as a color. | Object |
```

Comment:
The data field can also be used to stream additional things related to the cursor movement. Could be useful for annotations (not tried to see how this would work exactly but no harm throwing it out there for people to get creative until we add more info ourselves). Another caveat here is that some people might try to stream regular data here - worth highlighting that this represents delayed updates due to the special nature of this channel, so it's probably not suitable for streaming regular things.

Another use-case for data jfyi is adding id numbers of elements that may need to move along with the cursor (like a drag and drop feature) - similar to what Denis built in the hackathon with this collaborative puzzle app.